### ISSUE TITLE
Can't set class on li and a elements

### ISSUE THREAD

**Issue by svycka** [zendframework/zend-navigation#76]
when working with bootsrap4 it is not enough to set class on ul element. Maybe add methods like setULClass for li and a?

**Comment by froschdesign** [comment]
You can [add the classes](https://github.com/zendframework/zend-view/blob/master/src/Helper/Navigation/Menu.php#L588-L599) to the `li` element instead of the `a` element or you can use a [partial script](https://docs.zendframework.com/zend-navigation/helpers/menu/#rendering-a-custom-menu-using-a-partial-view-script). The last option should help you.

At the moment we will not add more methods to the view helper, because we will rework and simplify the whole rendering of the helpers.

**Comment by svycka** [comment]
@francisdaigle yes I could but look https://getbootstrap.com/docs/4.0/components/navbar/#supported-content
```
<ul class="navbar-nav mr-auto">
      <li class="nav-item active">
        <a class="nav-link" href="#">Home</a>
```
all `ul`, `li` and `a` has to have a custom class and if I change from a to li this still not help.
And yes I can use a partial script(that's what I do now), but that's annoying because I have to manage active page acl and everything myself I haven't tested yet other helpers but they may have problems as well

**Comment by svycka** [comment]
@froschdesign when this could happen I guess not any time soon?

**Comment by froschdesign** [comment]
@svycka 
I know the Bootstrap "Navbar" and all the classes.

> …I haven't tested yet other helpers but they may have problems as well

Please keep in mind zend-navigation was developed and released before Bootstrap or Foundation had a first release.

>  …but that's annoying because I have to manage active page acl…

[It's very simple.](https://docs.zendframework.com/zend-navigation/helpers/menu/#using-acls-with-partial-view-scripts)

> when this could happen I guess not any time soon?

Why are you so pessimistic? We are already working on it! (zend-expressive support first, then the rework of the view helpers)
In the meantime, I would extend the view helper, then you do not need a partial script.

**Comment by svycka** [comment]
I am not pessimistic just know that zend-* has a lot of projects and only a few people to work on them.

**Comment by weierophinney** [comment]
This repository has been closed and moved to laminas/laminas-navigation; a new issue has been opened at https://github.com/laminas/laminas-navigation/issues/1.
